Add a proxy with authentication option

It would be nice to allow Googler to query Google through an authenticated proxy.

At the moment, when I run something like `googler -p http://user:pwd@myproxy:port jarun` I get the following error:

```
[ERROR] Failed to connect to proxy server user:pwd@myproxy:port: [Errno -2] Name or service not known.
```

--- a/googler
+++ b/googler
@@ -671,23 +671,20 @@ class GoogleConnection(object):
 
         proxy = self._proxy
 
-        auth_needed = False
-        if proxy.find('@'):
-            auth_needed = True
-            proxy_host_port = proxy[proxy.find('@')+1:]
-            proxy_user_pwd = proxy[0:proxy.find('@')]
-            logger.debug(proxy_host_port)
-            logger.debug(proxy_user_pwd)
-
         if proxy:
+            auth_needed = False
+            if proxy.find('@'):
+                auth_needed = True
+                proxy_host_port = proxy[proxy.find('@')+1:]
+                proxy_user_pwd = proxy[0:proxy.find('@')]
+
             logger.debug('Connecting to proxy server %s', proxy)
             self._conn = TLS1_2Connection(proxy_host_port, timeout=timeout)
 
             logger.debug('Tunnelling to host %s' % host_display)
 
-            auth_header_value = 'Basic ' + base64.b64encode(proxy_user_pwd.encode('utf-8')).decode('utf-8')
-
             if auth_needed:
+                auth_header_value = 'Basic ' + base64.b64encode(proxy_user_pwd.encode('utf-8')).decode('utf-8')
                 self._conn.set_tunnel(host, port=port, headers={'Proxy-Authorization': auth_header_value})
             else:
                 self._conn.set_tunnel(host, port=port)

--- a/googler
+++ b/googler
@@ -18,6 +18,7 @@
 
 import argparse
 import atexit
+import base64
 import collections
 import codecs
 import functools
@@ -669,12 +670,27 @@ class GoogleConnection(object):
         host_display = host + (':%d' % port if port else '')
 
         proxy = self._proxy
+
+        auth_needed = False
+        if proxy.find('@'):
+            auth_needed = True
+            proxy_host_port = proxy[proxy.find('@')+1:]
+            proxy_user_pwd = proxy[0:proxy.find('@')]
+            logger.debug(proxy_host_port)
+            logger.debug(proxy_user_pwd)
+
         if proxy:
             logger.debug('Connecting to proxy server %s', proxy)
-            self._conn = TLS1_2Connection(proxy, timeout=timeout)
+            self._conn = TLS1_2Connection(proxy_host_port, timeout=timeout)
 
             logger.debug('Tunnelling to host %s' % host_display)
-            self._conn.set_tunnel(host, port=port)
+
+            auth_header_value = 'Basic ' + base64.b64encode(proxy_user_pwd.encode('utf-8')).decode('utf-8')
+
+            if auth_needed:
+                self._conn.set_tunnel(host, port=port, headers={'Proxy-Authorization': auth_header_value})
+            else:
+                self._conn.set_tunnel(host, port=port)
 
             try:
                 self._conn.connect(self._notweak)

